core/dbt/include/global_project/macros/materializations/incremental/incremental.sql

```diff
@@ -28,13 +32,23 @@
       {% set build_sql = create_table_as(False, intermediate_relation, sql) %}
       {% set need_swap = true %}
       {% do to_drop.append(backup_relation) %}
+  
   {% else %}
-      {% set tmp_relation = make_temp_relation(target_relation) %}
       {% do run_query(create_table_as(True, tmp_relation, sql)) %}
+      {% set schema_changed = check_for_schema_changes(tmp_relation, target_relation) %}
```

Comment:
This macro needs to run introspective queries, so it shouldn't execute if `on_schema_change = 'ignore'`—or if, per comment below, `on_schema_change = 'full_refresh'` and the schema has already been checked to trigger standard full-refresh logical pathway.

You could either or embed that check within these macros, or add it as an explicit check here in the materialization. I think I prefer the explicitness of having it here:
```sql
{% set tmp_relation = make_temp_relation(target_relation) %}
{% do run_query(create_table_as(True, tmp_relation, sql)) %}
{% if on_schema_change not in ('ignore', 'full_refresh') %}
  {% set schema_changed = check_for_schema_changes(tmp_relation, target_relation) %}
  {% do process_schema_changes(schema_changed, on_schema_change, tmp_relation, target_relation) %}
{% endif %}